lib/grpc_web/client/client_executor.rb

```diff
@@ -16,28 +16,28 @@ class << self
     GRPC_MESSAGE_HEADER = 'grpc-message'
     GRPC_HEADERS = %W[x-grpc-web #{GRPC_STATUS_HEADER} #{GRPC_MESSAGE_HEADER}].freeze
 
-    def request(uri, rpc_desc, params = {})
+    def request(uri, rpc_desc, params = {}, opt_headers = {})
       req_proto = rpc_desc.input.new(params)
       marshalled_proto = rpc_desc.marshal_proc.call(req_proto)
       frame = ::GRPCWeb::MessageFrame.payload_frame(marshalled_proto)
       request_body = ::GRPCWeb::MessageFraming.pack_frames([frame])
 
-      resp = post_request(uri, request_body)
+      resp = post_request(uri, request_body, opt_headers)
       resp_body = handle_response(resp)
       rpc_desc.unmarshal_proc(:output).call(resp_body)
     end
 
     private
 
-    def request_headers
+    def request_headers(opt_headers)
       {
         'Accept' => PROTO_CONTENT_TYPE,
         'Content-Type' => PROTO_CONTENT_TYPE,
-      }
+      }.merge(opt_headers)
     end
 
-    def post_request(uri, request_body)
-      request = Net::HTTP::Post.new(uri, request_headers)
+    def post_request(uri, request_body, opt_headers = {})
```

Comment:
Nitpick: since this is a private method and all calls to `post_request` include the trailing argument it can be a mandatory argument (instead of optional).